pom.xml

```diff
@@ -63,10 +63,30 @@
 
         <version.junit>4.12</version.junit>
         <version.org.easymock>3.5.1</version.org.easymock>
-        <version.io.takari.junit>1.2.7</version.io.takari.junit>
-        <version.com.h2database>1.4.178</version.com.h2database>
+        <version.org.unitils>3.4.6</version.org.unitils>
+        <version.simple-jndi>0.11.4.1</version.simple-jndi>
         <version.org.assertj.assertj-core>3.9.1</version.org.assertj.assertj-core>
         <version.org.skyscreamer.jsonassert>1.2.3</version.org.skyscreamer.jsonassert>
+        <version.com.github.tomakehurst.wiremock>2.5.1</version.com.github.tomakehurst.wiremock>
+        <!-- Fixes dependency convergence within Wiremock, see below -->
+        <version.com.fasterxml.jackson>2.8.6</version.com.fasterxml.jackson>
+        <version.org.jboss.byteman>4.0.4</version.org.jboss.byteman>
+        <version.io.takari.junit>1.2.7</version.io.takari.junit>
+        <version.com.h2database>1.4.178</version.com.h2database>
+
+        <!-- Integration tests -->
+
+        <version.org.openjdk.jmh>1.20</version.org.openjdk.jmh>
+        <version.org.jboss.arquillian>1.1.11.Final</version.org.jboss.arquillian>
+        <version.org.jboss.shrinkwrap.resolver>2.2.2</version.org.jboss.shrinkwrap.resolver>
+        <version.org.wildfly.arquillian>2.0.0.Final</version.org.wildfly.arquillian>
+        <version.org.springframework.boot>1.5.3.RELEASE</version.org.springframework.boot>
+        <version.org.apache.karaf>4.1.2</version.org.apache.karaf>
+        <version.org.ops4j.pax.exam>4.11.0</version.org.ops4j.pax.exam>
+        <version.org.ops4j.pax.url>1.6.0</version.org.ops4j.pax.url>
+        <!-- Necessary for OSGi integration tests with Karaf -->
+        <version.org.javassist>3.23.1-GA</version.org.javassist>
+        <version.org.jboss.jandex>2.0.5.Final</version.org.jboss.jandex>
```

Comment:
Done. I kept the reorganization of properties in the legacy module, though, because I had a hard time finding out where each dependency is really useful.